src/commands.ts

```diff
@@ -512,12 +513,30 @@ export const insertInlineFootnote =
         footnotesSection.node,
         schema.nodes.footnotes_element
       )
-      // TODO: Revisit this position calculation as it doesn't sound right to always push the note to the end.
+      // Get position for inserting the footnote
+      let index = 0
+      // TODO:: replace zero with body position
+      state.doc.slice(0, insertedAt).content.descendants((node) => {
```

Comment:
Would it be faster to use the footnotes plugin state to get the inline-footnotes, filter them based on the `insertedAt` position, and then use the length + 1 as the index? 🤔

Reply:
I have added comment